### PR TITLE
Added optional domain parameter

### PR DESCRIPTION
**Pull request by tomduijf**
Hi there,

This PR extends thee formula to to optionally add the ```domain:``` parameter if set in the pillar.

### CODE REVIEW

File: resolver/init.sls [hunk 1]
```diff
@@ -21,6 +21,7 @@ resolv-file:
         nameservers: {{ salt['pillar.get']('resolver:nameservers', ['8.8.8.8','8.8.4.4']) }}
         searchpaths: {{ salt['pillar.get']('resolver:searchpaths', [salt['grains.get']('domain'),]) }}
         options: {{ salt['pillar.get']('resolver:options', []) }}
+        domain: {{ salt['pillar.get']('resolver:domain') }}
```
Comment (gravyboat) [inline resolver/init.sls]: Could you please add a default value here?

**Comment by tomduijf** [comment]
It's an optional parameter, must be explicitly set in pillar.
Setting a default will result in unwanted client behaviour.

File: resolver/init.sls (hunk 1 above)
Comment (tomduijf) [inline resolver/init.sls]: It's an optional parameter, must be explicitly set in pillar.
Setting a default will result in unwanted client behaviour.

**Comment by gravyboat** [comment]
Gotcha, I forgot that's how this formula operated. Thanks.

**Pull request merged**